### PR TITLE
README updates.

### PR DESCRIPTION
**Pull request by sebastianv1**
Easier introduction into Cleanse (maintain current setup guide but moves into an "Advanced Setup" section), and cleans up many incorrect APIs and stale information.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -424,23 +341,30 @@ Consuming the Primary API URL (e.g. "https://connect.squareup.com/v2/")
...
-is what's returned when a ``Component`` is constructed. It also may be referred to as an "entry point",
+Components
+`````````````````````
+Cleanse has a concept of a ``Component``. A ``Component`` represents an object graph of our dependencies that returns the ``Root`` `associated type`_ upon construction and is used as the "entry point" into Cleanse. However, we can also use a ``Component`` to create a subgraph inside our parent object graph, called a subcomponent. Subcomponents are closely related to scopes_ and are used to scope your dependencies. Objects inside a component are only allowed to inject dependencies that exist within the same component (or scope), or an ancestor's component. A parent component is not allowed to reach into a subcomponent and retrieve a dependency. One example of using components to scope dependencies is by having a ``LoggedInComponent`` inherting from your application's Root component. This allows you to bind logged in specific objects such as session tokens or account objects within the ``LoggedInComponent`` so that you can't accidently leak these dependencies into objects used outside of a logged session (i.e welcome flow views).
```
Comment (holmes) [inline README.rst]: `Subcomponents are closely related to scopes_ and are used to scope` what's with the `scopes_`/ Supposed to be link?

File: README.rst [hunk 2]
```diff
@@ -424,23 +341,30 @@ Consuming the Primary API URL (e.g. "https://connect.squareup.com/v2/")
...
+      associatedtype Scope: Cleanse._ScopeBase = Unscoped
+
+      static func configure(binder: Binder<Self.Scope>)
+
+      static func configureRoot(binder bind: ReceiptBinder<Root>) -> BindingReceipt<Root>
     }
 
-The outermost component of an object graph (e.g. the Root component), is built by the ``build()`` method.
+
+The outermost component of an object graph (e.g. the Root component), is built by the ``build(())`` method on `ComponentFactory`.
```
Comment (holmes) [inline README.rst]: Another too many ()'s

File: README.rst [hunk 3]
```diff
@@ -477,10 +401,35 @@ Defining a component
...
-    root.somethingUsingTheAPI.doSomethingFun()
+  struct Root : RootComponent {
+      func configure<B : Binder>(binder binder: B) {
+          binder.install(dependency: APIComponent.self)
+      }
+      // ...
+  }
+  
+
+And then you can use it by injecting in the ``ComponentFactory<APIComponent>`` instance into an object and calling ``build(())``.
```
Comment (holmes) [inline README.rst]: parens

File: README.rst [hunk 4]
```diff
@@ -535,22 +482,19 @@ Tag Step (Optional)
 An optional step that indicates that the provided type should actually be
 ``TaggedProvider<Element>`` and not just ``Provider<Element>``.
 
-.. seealso::
-
-  `Type Tags`_ for more information
+See: `Type Tags`_ for more information
 
+.. _scopes:
```
Comment (holmes) [inline README.rst]: oh, here's the link

File: README.rst [hunk 5]
```diff
@@ -535,22 +482,19 @@ Tag Step (Optional)
...
 
 Scope Step
 ~~~~~~~~~~~~~~~~~~~~~
 
 By default, whenever an object is requested, Cleanse constructs a new one.
 If the optional `.sharedInScope()` is specified, Cleanse will memoize and return the same instance in the scope of the ``Component``
-it was configured in. So if this is configured as a singleton in the `RootComponent`, then will return the same instance for the entire app.
+it was configured in. Each ``Component`` requires its own `Scope` type. So if this is configured as a singleton in the `RootComponent`, then will return the same instance for the entire app.
```
Comment (holmes) [inline README.rst]: join w/ the line above

File: README.rst [hunk 6]
```diff
@@ -687,10 +631,7 @@ Cleanse will only allow one to provide elements into an ``Array``. The choice of
 Java where every type of object can be part of a ``Set``, only types that are ``Hashable`` can be part of a ``Set``
```
Comment (holmes) [inline README.rst]: This sentence is missing a word. Maybe start w/ `Unlike`?

Comment (holmes) [reply README.rst]: Actually this whole chunk could use some love.

File: README.rst [hunk 7]
```diff
@@ -75,78 +76,74 @@ one must import it.
...
 .. code-block:: swift
 
-  extension AppDelegate {
-    struct Component : Cleanse.RootComponent {
-      // When we call build() it will return the Root type, which is a PropertyInjector<AppDelegate>.
-      // More on how we use the PropertyInjector type later.
-      typealias Root = PropertyInjector<AppDelegate>
-      
+  struct Component : Cleanse.RootComponent {
+      // When we call build(()) it will return the Root type, which is a RootViewController instance.
```
[the remark this answers is not in the document]
Comment (sebastianv1) [reply README.rst]: Nah it's correct. `Void=()` since Swift 4.2

[the remark this answers is not in the document]
Comment (holmes) [reply README.rst]: Well aren't I just out of the loop.

**Pull request merged**